### PR TITLE
Update for Microsoft build of OpenJDK 17, Jan 2022 PSU

### PR DESCRIPTION
**Pull request by d3r3kk**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?
-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/44370)

### CODE REVIEW

File: manifests/m/Microsoft/OpenJDK/17/17.0.2.8/Microsoft.OpenJDK.17.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+InstallModes:
+- interactive
+- silent
+- silentWithProgress
+InstallerSwitches:
+  Custom: ADDLOCAL=FeatureMain,FeatureEnvironment,FeatureJarFileRunWith,FeatureJavaHome
+UpgradeBehavior: uninstallPrevious
+Installers:
+- Architecture: x64
+  InstallerUrl: https://aka.ms/download-jdk/microsoft-jdk-17.0.2.8.1-windows-x64.msi#winget
```
Comment (Trenly) [inline manifests/m/Microsoft/OpenJDK/17/17.0.2.8/Microsoft.OpenJDK.17.installer.yaml:18]:
```suggestion
  InstallerUrl: https://download.visualstudio.microsoft.com/download/pr/9b41bfa3-0cc3-4830-aef4-5b861199d54d/c30c8279b083c2fe50ab70e2d0def72a/microsoft-jdk-17.0.2.8.1-windows-x64.msi
```

File: manifests/m/Microsoft/OpenJDK/17/17.0.2.8/Microsoft.OpenJDK.17.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  InstallerSha256: 413C0354CB2F231405C8D2BE48847A4499050C15A7F01C3C1F42C6D84F059EE8
+  ProductCode: '{A34F9424-5651-47B0-825B-9C761BB7430B}'
+- Architecture: arm64
+  InstallerUrl: https://aka.ms/download-jdk/microsoft-jdk-17.0.2.8.1-windows-aarch64.msi#winget
```
Comment (Trenly) [inline manifests/m/Microsoft/OpenJDK/17/17.0.2.8/Microsoft.OpenJDK.17.installer.yaml:22]:
```suggestion
  InstallerUrl: https://download.visualstudio.microsoft.com/download/pr/9b41bfa3-0cc3-4830-aef4-5b861199d54d/2eeac5626a460f38a8e223beba9e1275/microsoft-jdk-17.0.2.8.1-windows-aarch64.msi
```

**Review by Trenly — CHANGES_REQUESTED** [review]

File: manifests/m/Microsoft/OpenJDK/17/17.0.2.8/Microsoft.OpenJDK.17.installer.yaml (hunk 1 above)
Comment (d3r3kk) [reply manifests/m/Microsoft/OpenJDK/17/17.0.2.8/Microsoft.OpenJDK.17.installer.yaml:18]: Leaving as is, we use aka.ms to measure usage.

File: manifests/m/Microsoft/OpenJDK/17/17.0.2.8/Microsoft.OpenJDK.17.installer.yaml (hunk 2 above)
Comment (d3r3kk) [reply manifests/m/Microsoft/OpenJDK/17/17.0.2.8/Microsoft.OpenJDK.17.installer.yaml:22]: Leaving as-is, we use aka.ms to measure usage.

**Review by OfficialEsco — APPROVED** [review]

**Pull request merged**